ImageAnalysis/image_analysis/base.py

```diff
@@ -191,14 +180,20 @@ def apply_background(self, background):
         self.background = background
 
     def configure(self, **kwargs):
-        """ Configures necessary parameters for the analyzer used by analyze_image
+        """ Given a dictionary of keyword arguments, updates environment variables for the anlayzer
 
-        To be implemented by children of this class.  This is called by the initialization function, where the keyword
-        arguments are given by either the "settings" block of the .ini config file or explicitly listed.
+        This function also requires that the class variables were previously initialized to their proper type.
+        Furthermore, passing in a None for a given keyword argument will skip over resetting the variable.
```

Comment:
Another way, which doesn't require that variables have the correct class*, is to get the expected types from the `__init__` method's annotations, using `typing.get_type_hints(image_analyzer_instance_or_class.__init__)`

\* This isn't guaranteed even with coercion in `__init__`, because users can still manually change the attributes from outside (`image_analyzer_instance.some_parameter = 1.23`).